dotCMS/src/main/java/com/dotcms/graphql/business/GraphqlAPIImpl.java

```diff
@@ -160,137 +97,37 @@ private void printSchema() {
         }
     }
 
-    private void createSchemaType(ContentType contentType,
-                                              final Map<String, GraphQLObjectType> graphqlObjectTypes) {
-
-        // skip contentType.variable not sticking to the regex
-        if(!contentType.variable().matches(TYPES_AND_FIELDS_VALID_NAME_REGEX)) {
-            return;
-        }
-
-        final GraphQLObjectType.Builder builder = GraphQLObjectType.newObject().name(contentType.variable());
-
-        // add CONTENT interface fields
-        builder.fields(InterfaceType.CONTENTLET.getType().getFieldDefinitions());
-
-        if(InterfaceType.getInterfaceForBaseType(contentType.baseType())!=null) {
-            builder.withInterface(InterfaceType.getInterfaceForBaseType(contentType.baseType()));
-        }
-
-        final List<Field> fields = contentType.fields();
-
-        fields.forEach((field)->{
-            // skip field.variable not sticking to the regex
-            if(!field.variable().matches(TYPES_AND_FIELDS_VALID_NAME_REGEX)
-                || field instanceof RelationshipsTabField) {
-                return;
-            }
+    @LogTime(loggingLevel = "INFO")
+    private GraphQLSchema generateSchema() throws DotDataException {
+        final Set<GraphQLType> graphQLTypes = new HashSet<>();
 
-            if(!(field instanceof RowField) && !(field instanceof ColumnField)) {
-                if (field instanceof RelationshipField) {
-                    try {
-                        handleRelationshipField(contentType, builder, field, graphqlObjectTypes);
-                    } catch(DotStateException e) {
-                        Logger.error(this, "Unable to create relationship field", e);
-                    }
-                } else {
-                    builder.field(newFieldDefinition()
-                        .name(field.variable())
-                        .type(field.required()
-                            ? nonNull(getGraphqlTypeForFieldClass(field.type(), field))
-                            : getGraphqlTypeForFieldClass(field.type(), field))
-                        .dataFetcher(getGraphqlDataFetcherForFieldClass(field.type()))
-                    );
-                }
+        for (GraphQLTypesProvider typesProvider : typesProviders) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Local variable 'typesProvider' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5642169)